src/dmd/cpreprocess.d

```diff
@@ -33,11 +33,12 @@ import dmd.root.string;
  * Preprocess C file.
  * Params:
  *      csrcfile = C file to be preprocessed, with .c or .h extension
+ *      ifile = set to true if an output file was written
  * Result:
  *      filename of output
  */
 extern (C++)
-FileName preprocess(FileName csrcfile)
+FileName preprocess(FileName csrcfile, out bool ifile)
```

Comment:
Perhaps, though I could just stash the contents into `global.fileManager` so that the subsequent `fileManager.lookup` returns the content without needing to touch the disk.

Saying that, I also suspect that libcpp will mostly parse the entire C source file for me, so there might have to be some other chicanery done in the background (fix-up in Compiler.onParseModule maybe).

Reply:
Looking a bit further into the cpplib interface, there's a bunch of callbacks that I can register as the library preprocesses the file (i.e: file_change, line_change, include, define, undef, etc...) which I can capture and possibly format/write out to a temporary file.

It would still be nice to stream-line this into CParser somehow, though I can make do with an intermediate for now.

Still need to sort out the implementation of mixinC though, and what to do with module conflicts caused by flattened headers. ;-)

Reply:
Could just replace it with an interface properly